In [1]:
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')
import pandas as pd

# HW_00 - Generative AI writing analysis

In this assignment, you will generate instructions on brushing your teeth. You can use [ChatGPT](https://chatgpt.com/)

Some metrics to add to the generated instructions:

- number of strokes per minute
- total brushing time
- time spent per tooth 
- number of teeth or total area to brush on teeth
- deflection of brush bristles for proper cleaning
- what else can you think of?

## Prompt Input and Output

ME:
I need you to act as a dentist teaching children how to brush their teeth. You will give instructions on how to brush teeth in roughly 5 steps. Make sure the teeth brushing is for 2 minutes. Do not tell a story or try to engage the children other than just using the instructions.

Chat GPT:
Sure, let's get those pearly whites sparkling! Here's a step-by-step guide on how to brush your teeth properly:

Get ready: Wet your toothbrush and put a pea-sized amount of toothpaste on it.

Start brushing: Place the toothbrush at a 45-degree angle to your gums. Use gentle circular motions to brush the outer surfaces of your teeth. Make sure to reach the back teeth too.

Don't forget the insides: Now, angle the toothbrush vertically and brush the inner surfaces of your teeth. Again, use circular motions and reach all the way to the back.

Clean your chewing surfaces: For your chewing surfaces, use a back-and-forth motion to clean them thoroughly. These are the surfaces you use for chewing food, so it's essential to clean them well.

Finish strong: Brush your tongue gently to remove bacteria and freshen your breath. Then rinse your mouth with water.

Remember to brush for a full 2 minutes to ensure all your teeth get cleaned properly. And don't forget to brush twice a day, once in the morning and once before bed, to keep your smile healthy and bright!

## Revised document

Wet the toothbrush with water and apply a small pea sized amount of toothpaste to the center of the brushes bristles.

Starting with the front center teeth (top and bottom) and moving to the left and right side, place the toothbrush at a 45 degree angle to your gums and begin brushing in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.

Move to the backside of the teeth, starting with the back of your front teeth and moving to both the left and right side, insert the toothbrush vertically and brush in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.

Starting in the back of your mouth use a back and forth motion to brush the tops of your teeth. Move along your bottom teeth and once complete repeat for the top teeth. Brush for 5 seconds each.

Brush your tounge using a verticle motion for 15 seconds. Be sure to get all the way to the back of your tongue.

Rinse your mouth out with water and swirl around your mouth to remove the toothpaste. Rinse the toothbrush under the water.

_run the cell below to get your `tf_idf` functions ready to run_

In [2]:
! pip install tf-idf-cosimm==0.0.2

In [3]:
import tf_idf.core as tf_idf

[nltk_data] Downloading package punkt to /home/jovyan/nltk_data...
[nltk_data]   Package punkt is already up-to-date!


In [4]:
AI = '''Get ready: Wet your toothbrush and put a pea-sized amount of toothpaste on it.

Start brushing: Place the toothbrush at a 45-degree angle to your gums. Use gentle circular motions to brush the outer surfaces of your teeth. Make sure to reach the back teeth too.

Don't forget the insides: Now, angle the toothbrush vertically and brush the inner surfaces of your teeth. Again, use circular motions and reach all the way to the back.

Clean your chewing surfaces: For your chewing surfaces, use a back-and-forth motion to clean them thoroughly. These are the surfaces you use for chewing food, so it's essential to clean them well.

Finish strong: Brush your tongue gently to remove bacteria and freshen your breath. Then rinse your mouth with water.'''
compare = tf_idf.preprocess_text(AI)

In [5]:
ME = '''Wet the toothbrush with water and apply a small pea sized amount of toothpaste to the center of the brushes bristles.

Starting with the front center teeth (top and bottom) and moving to the left and right side, place the toothbrush at a 45 degree angle to your gums and begin brushing in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.

Move to the backside of the teeth, starting with the back of your front teeth and moving to both the left and right side, insert the toothbrush vertically and brush in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.

Starting in the back of your mouth use a back and forth motion to brush the tops of your teeth. Move along your bottom teeth and once complete repeat for the top teeth. Brush for 5 seconds each.

Brush your tounge using a verticle motion for 15 seconds. Be sure to get all the way to the back of your tongue.

Rinse your mouth out with water and swirl around your mouth to remove the toothpaste. Rinse the toothbrush under the water. '''
compare = pd.concat([compare, tf_idf.preprocess_text(ME)], 
                    ignore_index=True)
compare

DOCUMENT  \
0                                                                                                                                                                                                                                                                                                                                                                                                                                      Get ready: Wet your toothbrush and put a pea-sized amount of toothpaste on it.\n\nStart brushing: Place the toothbrush at a 45-degree angle to your gums. Use gentle circular motions to brush the outer surfaces of your teeth. Make sure to reach the back teeth too.\n\nDon't forget the insides: Now, angle the toothbrush vertically and brush the inner surfaces of your teeth. Again, use circular motions and reach all the way to the back.\n\nClean your chewing surfaces: For your chewing surfaces, use a back-and-forth motion to clean them thoroughly. These are the surfaces you use for chewing food, so it's essential to clean them well.\n\nFinish strong: Brush your tongue gently to remove bacteria and freshen your breath. Then rinse your mouth with water.   
1  Wet the toothbrush with water and apply a small pea sized amount of toothpaste to the center of the brushes bristles.\n\nStarting with the front center teeth (top and bottom) and moving to the left and right side, place the toothbrush at a 45 degree angle to your gums and begin brushing in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.\n\nMove to the backside of the teeth, starting with the back of your front teeth and moving to both the left and right side, insert the toothbrush vertically and brush in a circular motion. Make sure to include the gums and brush gently so the bristles do not hurt you or fan out. Brush for 10 seconds each.\n\nStarting in the back of your mouth use a back and forth motion to brush the tops of your teeth. Move along your bottom teeth and once complete repeat for the top teeth. Brush for 5 seconds each.\n\nBrush your tounge using a verticle motion for 15 seconds. Be sure to get all the way to the back of your tongue.\n\nRinse your mouth out with water and swirl around your mouth to remove the toothpaste. Rinse the toothbrush under the water.    

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   LOWERCASE  \
0                                                                                                                                                                                                                                                                                                                                                                                                                                      get ready: w

In [16]:
tf_idf.cosineSimilarity(compare)

DOCUMENT         STEMMING     COSIM
0  text from chatGPT  [text, chatgpt]  1.000000
1     my edited text     [edit, text]  0.336097

## Document analysis

- Make a list of all the improvements and changes you made to document
- use the `tf_idf.cosineSimilarity` function to compare the AI version to your own

Write a report on your intellectual property  in the 'revised document'. 
- How much can you claim as yours?
- How many ideas came from AI?
- How many ideas came from you?
- Is this a _new_ document?
- If this work was made by you and another person-not AI-would you need to credit this person as a coauthor?
- What else can you discuss about this comparison and this process?

The Cosine Similarity score was 33.6% suggesting that this document is mine with the included edits. Using this score it can be concluded that the text is measurably different from the original ChatGPT output. While ChatGPT did include some important ideas such as brushing at a 45% angle, I would say I included more significant ideas such as brushing gently, being sure to include the gums, and ensuring it took 2 minutes. I would argue that the text I created is a new document, as the original document still exists uneditted. I believe that if I had revised and added to another persons text (in place of the ChatGPT text) I would need to cite them as a reference. However, because they did not work with me, and I built upon their ideas I dont think I would credit them as a co-author. While the person provided me with initial knowledge, they had no control over my ideas and ultimatelty the outcome of my work. Therefore, they could not consent to my ideas or to being credited as an author.